lib/segment/src/data_types/vectors.rs

```diff
@@ -44,10 +149,10 @@ impl Validate for Vector {
 }
 
 impl<'a> VectorRef<'a> {
-    pub fn to_vec(self) -> Vector {
+    pub fn to_vec(self) -> internal::Vector {
```

Comment:
`VectorRef` is related to internal vector, not external. I didn't allow to take `VectorRef` from `Vector`, only from `internal::Vector`